### PR TITLE
feat: add generate component command

### PR DESCRIPTION
**Pull request by brandonlilly**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [x] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [x] Tested my solution on Mobile & Tablet.
- [x] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [x] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [x] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [x] Made sure that all accessibility errors are resolved.
- [x] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [x] Added name to OWNERS file for all new components
- [x] If adding a new component, add its export to the rollup config
- [x] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

Adds a `yarn generate MyComponent` command to generate a new component from a template.

### CODE REVIEW

File: scripts/generateComponent.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        // copy template component
+        fs.copySync(srcPath, destPath);
+
+        // replace text in files
+        replace.sync({
+            files,
+            from: /MyComponent/g,
+            to: componentName,
+        });
+        replace.sync({
```
Comment (zero0Halo) [inline scripts/generateComponent.js]: I'm fuzzy with what's going on with this instance of `replace.sync`. I commented it out and the script appeared to work the same.

File: scripts/MyComponent/MyComponent.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    font-family: ${th('typography.fontFamily')};
+`;
+
+storiesOf('Components/MyComponent', module).add('Default', () => (
+    <ThemeProvider theme={RootTheme}>
+        <StyledStory>
+            <MyComponent />
+        </StyledStory>
+    </ThemeProvider>
+));
```
Comment (zero0Halo) [inline scripts/MyComponent/MyComponent.stories.tsx:24]: I think it'd be good to include at least one knob so the dev can expand on it.

File: scripts/generateComponent.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+const fs = require('fs-extra');
+const argv = require('yargs').argv;
+const replace = require('replace-in-file');
+const Renamer = require('renamer');
+const renamer = new Renamer();
+
+const componentName = argv._[0];
+
+if (componentName) {
```
Comment (zero0Halo) [inline scripts/generateComponent.js]: In addition to checking if an argument was passed for the component name, it might also be good to have a check to ensure that the component doesn't already exist. Otherwise this script will still generate files in the destination folder. Attaching a screenshot of the two components I made, then running the script again. 

image

**Review by zero0Halo — COMMENTED** [review]
I think an entry in the contributors page would be awesome. This is a great tool to help people get started.

File: scripts/MyComponent/MyComponent.stories.tsx (hunk 2 above)
Comment (brandonlilly) [reply scripts/MyComponent/MyComponent.stories.tsx:24]: I don't really want to do this because I'd want the knob to be for a prop, and there is no prop I'd want by default.

**Review by zero0Halo — APPROVED** [review]
